builder.go

```diff
@@ -562,24 +562,51 @@ var builtinAllowedBuildArgs = map[string]bool{
 	"no_proxy":    true,
 }
 
-// ParseDockerIgnore returns a list of the excludes in the .dockerignore file.
+// ParseIgnore returns a list of the excludes in the specified path
+// path should be a file with the .dockerignore format
 // extracted from fsouza/go-dockerclient and modified to drop comments and
 // empty lines.
-func ParseDockerignore(root string) ([]string, error) {
+func ParseIgnore(path string) ([]string, error) {
 	var excludes []string
-	ignore, err := ioutil.ReadFile(filepath.Join(root, ".dockerignore"))
-	if err != nil && !os.IsNotExist(err) {
-		return excludes, fmt.Errorf("error reading .dockerignore: '%s'", err)
+
+	ignores, err := ioutil.ReadFile(path)
+	if err != nil {
+		return excludes, err
 	}
-	for _, e := range strings.Split(string(ignore), "\n") {
-		if len(e) == 0 || e[0] == '#' {
+	for _, ignore := range strings.Split(string(ignores), "\n") {
+		if len(ignore) == 0 || ignore[0] == '#' {
 			continue
 		}
-		excludes = append(excludes, e)
+		// drop  "/" when begining or ending an ignore string
+		for {
+			if !strings.HasPrefix(ignore, "/") {
+				break
+			}
+			ignore = strings.TrimPrefix(ignore, "/")
+		}
+		for {
+			if !strings.HasSuffix(ignore, "/") {
+				break
+			}
+			ignore = strings.TrimSuffix(ignore, "/")
+		}
+		excludes = append(excludes, ignore)
```

Comment:
Maybe use `strings.Trim()` here, and guard against the end-result of the trimming being an empty string.

Reply:
Nice, I did not know this was available.